tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -105,6 +106,10 @@ type Store interface {
 
 type storeImpl struct {
     db *pgxpool.Pool
+
+    // Lock since copyFrom requires a delete first before being executed we can get in odd states if
+    // multiple processes are trying to work on the same subsets of rows.
```

Comment:
The mutex is available globally. You can move this specific comment to UpsertMany before locking.